### PR TITLE
Feature/703 ephemeris based guidance

### PR DESCRIPTION
**Pull request by JulianHammerl**
* **Tickets addressed:** MAXGNC-703
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

A module was created that takes the ephemeris message containing the position and velocity of two objects (a base object and a secondary object), and computes the relative position and velocity of the secondary with respect to the base. This is then written to a navigation and filter message.

## Verification

A UnitTest was added.

## Documentation

An rst documentation was added.

## Future work

N/A

### CODE REVIEW

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (thibaudteil) [inline src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp:4]: I've started putting 2024, University of Colorado at Boulder
worth confirming

**Review by thibaudteil — APPROVED** [review]

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (patkenneally) [inline src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp]: :large_blue_circle: This reads weird. I know it's just a copy paste. What do you think of "This method resets the module state to its initialized default."

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (patkenneally) [inline src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp]: :large_orange_diamond: This comment is redundant and can be removed.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h
Comment (patkenneally) [inline src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h]: :large_orange_diamond: For both of these methods you can adjust `CurrentSimNanos` to `currentSimNanos` and mark the methods as overridden. E.g. `void UpdateState(uint64_t currentSimNanos) override`

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (patkenneally) [inline src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp]: :large_orange_diamond: Change to camelCase.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h
Comment (patkenneally) [inline src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h]: :small_red_triangle_down: We can make the return type on both of these getters non-const. The const has no effect on the return value.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (patkenneally) [inline src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp]: :large_blue_circle: What is the base object? Can we say what the object is rather than using the word object. Similar further down.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h
Comment (patkenneally) [inline src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h:25]: :small_red_triangle_down: This include is unused.

**Review by patkenneally — APPROVED** [review]
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

Nicely done.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp:4]: @patkenneally I thought we should use LASP?

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp]: Done.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp]: Done.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h
Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h:25]: Removed it.

Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h:25]: It is indicated as unused on my computer as well, but I need it to read and write the messages (e.g. cArray2EigenVector3d)?

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h
Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h]: Done.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp]: Replaced it with "celestial object" and added example of asteroid for base and example of spacecraft for secondary.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h
Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.h]: Done.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp]: Done.

File: src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp
Comment (patkenneally) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp:4]: This line is the author. In this case that should be LASP, Uni of Co Bol... etc. University of Colorado is also correct just less specific maybe?

Comment (JulianHammerl) [reply src/fswAlgorithms/transDetermination/ephemDifferenceWithUncertainty/ephemDifferenceWithUncertainty.cpp:4]: Okay, I will leave it this way then.

**Pull request merged**
